src/hooks/usePollTableQuery.ts

```diff
@@ -0,0 +1,119 @@
+} from "apollo-client/core/networkStatus";
+import { usePrevious } from "hooks";
+import { useParams, useLocation } from "react-router-dom";
+import { ApolloQueryResult } from "apollo-client";
+import isEqual from "lodash.isequal";
+
+export const usePollTableQuery = <ApolloQueryVariables, ApolloQueryResultType>({
+  networkStatus,
+  getQueryVariables,
+  refetch,
+  search,
+}: {
+  networkStatus: NetworkStatus;
+  getQueryVariables: (search: string, id?: string) => ApolloQueryVariables;
+  refetch: (
+    variables?: ApolloQueryVariables
+  ) => Promise<ApolloQueryResult<ApolloQueryResultType>>;
+  search: string;
+}): { showSkeleton: boolean } => {
+  const { pathname } = useLocation();
+  const [initialPathname] = useState(pathname);
+  const { id: resourceId } = useParams<{ id: string }>();
+  const [intervalId, setIntervalId] = useState<number>();
+  const [queryVarDiffOccured, setQueryVarDiffOccured] = useState(false);
+  const currentQueryVariables = getQueryVariables(search, resourceId);
+  const prevQueryVariables = usePrevious(currentQueryVariables);
+  const isLoading = isNetworkRequestInFlight(networkStatus);
+
+  useEffect(() => {
+    if (!isEqual(currentQueryVariables, prevQueryVariables)) {
+      setQueryVarDiffOccured(true);
+    }
+  }, [currentQueryVariables, prevQueryVariables, setQueryVarDiffOccured]);
+
+  useEffect(() => {
+    if (!isLoading) {
+      setQueryVarDiffOccured(false);
+    }
+  }, [isLoading, setQueryVarDiffOccured]);
+
+  useEffect(() => {
+    if (!intervalId) {
+      pollQuery({
+        setIntervalId,
+        refetch,
+        search,
+        getQueryVariables,
+        resourceId,
+      });
+    }
+    return () => clearInterval(intervalId);
+  }, [
+    intervalId,
+    refetch,
+    search,
+    getQueryVariables,
+    resourceId,
+    setIntervalId,
+  ]);
+
+  useEffect(() => {
+    if (pathname !== initialPathname) {
+      clearInterval(intervalId);
+    }
+  }, [pathname, initialPathname, intervalId]);
+
+  useEffect(() => {
+    if (
+      intervalId &&
+      !isEqual(currentQueryVariables, prevQueryVariables) &&
+      pathname === initialPathname
+    ) {
+      clearInterval(intervalId);
+      refetch(currentQueryVariables);
+      pollQuery({
+        search,
+        resourceId,
+        refetch,
+        setIntervalId,
+        getQueryVariables,
+      });
+    }
+  }, [
+    currentQueryVariables,
+    prevQueryVariables,
+    networkStatus,
+    refetch,
+    intervalId,
+    resourceId,
+    getQueryVariables,
+    search,
+    pathname,
+    initialPathname,
+  ]);
+
+  return {
+    showSkeleton: queryVarDiffOccured,
+  };
+};
+
+const pollInterval = 5000;
+
+const pollQuery = ({
+  search,
+  resourceId,
+  refetch,
+  setIntervalId,
+  getQueryVariables,
+}) => {
+  const queryVariables = getQueryVariables(search, resourceId);
+  const intervalId = window.setInterval(() => {
+    refetch(queryVariables);
+  }, pollInterval);
+  setIntervalId(intervalId);
+};
```

Comment:
why is `setInterval` being used? i am unable to understand its objective from reading the code in `usePollTableQuery`

Reply:
setInterval calls passed in function after a set amount of time. in this case the callback is a refetch function. so this is how the query is being polled hence the function name `pollQuery`

Reply:
are we unable to use the pollInterval via the query? i'm confused because we poll other things without using setInterval. Can you explain to me please why this implementation requires it?

Reply:
the pollInterval option doesn't allow you to specify new query variables from the original query

Reply:
i see. can we add a comment? this is a perfect example of a commenting answering _the why_. The dev reading this a year from now (might even be you) will thank past Arjun for the comment.